File: 1844.replace-all-digits-with-characters.js
/*
 * @lc app=leetcode id=1844 lang=javascript
 *
 * [1844] Replace All Digits with Characters
 */

// @lc code=start
/**
 * @param {string} s
 * @return {string}
 */
var replaceDigits = function (s) {
    let s_arr = s.toString().split('')
    for (let i in s_arr) {
        if (s_arr[i] >= '0' && s_arr[i] <= '9') {
            s_arr[i] = String.fromCharCode(s_arr[i - 1].charCodeAt(0) + +s_arr[i]);
        }
    }
    return s_arr.join('');
};
// @lc code=end

File: 1844.replace-all-digits-with-characters.cpp
/*
 * @lc app=leetcode id=1844 lang=cpp
 *
 * [1844] Replace All Digits with Characters
 */
#include <string>
using namespace std;
// @lc code=start
class Solution
{
public:
    string replaceDigits(string s)
    {
        char prev;
        for (auto &c : s)
        {
            if (isdigit(c))
            {
                c = c - '0' + prev;
            }
            else
            {
                prev = c;
            }
        }
        return s;
    }
};
// @lc code=end
